### ISSUE TITLE
.find() function does not work after a single use

### ISSUE THREAD

**Issue by avi-perl** [fredysomy/pysonDB#34]
**Describe the bug**
When writing data to the JSON db, after the first write to the file, the `.find()` method will return the record based on the ID. After adding a 2nd record and trying again, the `IdNotFoundError` error is thrown despite the record being in the DB!

**How to reproduce**
python 3.9.6
```php
from pysondb import db

if __name__ == "__main__":
    db = db.getDb("test.json")
    while True:
        id = db.add({"dev": "test"})
        print(db.find(id))
```

**Expected behavior**
The record should be returning as expected.

---

This is a neat library, great work!

**Comment by avi-perl** [comment]
Line 215 of db.py, the first instance of `raise IdNotFoundError(pk)` will raise after looking at the first item in the DB.

```python
    def find(self, pk :int, objectify: bool = False) -> getType:
        with self.lock:
            try:
                with open(self.filename, "r", encoding="utf8") as db_file:
                    db_data = self._get_load_function()(db_file)
                for d in db_data["data"]:
                    if(d[self.id_fieldname])==self._cast_id(pk):
                        return (
                        d
                        if not objectify
                        else self._objectify(json.dumps({"data": d})).data
                        )
                    else:
                        raise IdNotFoundError(pk)

            except:
                raise IdNotFoundError(pk)
```

**Comment by fredysomy** [comment]
Hi @avi-perl . Will get back to this issue , sort of busy now 😅

**Comment by LyndonFan** [comment]
Hi, is the issue still open? I can try to work on it!

**Comment by fredysomy** [comment]
Yes , this issue is still open and you can work on it, shall I assign it to you?

**Comment by LyndonFan** [comment]
That would be great, thanks!

**Comment by fredysomy** [comment]
Sure. Thanks , 🎉🎉

**Comment by fredysomy** [comment]
@avi-perl , this issue is resolved in #47 .

Closing it.
